src/components/Layout.jsx

```diff
@@ -1,15 +1,25 @@
-import React from 'react'
-
-import {Home} from '../pages'
+import React, { useState } from 'react'
+import {Home, ProductList} from '../pages'
 import {Footer, Navbar} from './index'
 
 const Layout = () => {
+
+  const [pageToRender, setPageToRender] = useState('Home')
+
+  const handleNavigation = (page) => {
+    setPageToRender(page)
+  }
+
   return (
     <div className='page-container'>
-        <Navbar />
+        <Navbar navigate={pageToRender !== 'Home' ? handleNavigation : null} />
```

Comment:
This may be a good use case for useContext, to avoid this prop, is not too intuitive to have this handler here